lib/matplotlib/tests/test_axes.py

```diff
@@ -2704,6 +2704,29 @@ def test_scatter_linewidths(self):
         assert_array_equal(pc.get_linewidths(),
                            [*range(1, 5), mpl.rcParams['lines.linewidth']])
 
+    def test_scatter_singular_plural_arguments(self):
+        try:
+            plt.scatter([1, 2, 3], [1, 2, 3], linewidths=[0.5, 0.4, 0.3], linewidth=0.2)
+            pytest.fail("Expected TypeError")
+        except TypeError:
+            pass
```

Comment:
Thanks for adding tests!

This is the typical way we test error:
```suggestion
        with pytest.raises(TypeError, match="linewidths and linewidth cannot be"):
            plt.scatter([1, 2, 3], [1, 2, 3], linewidths=[0.5, 0.4, 0.3], linewidth=0.2)
```